python/dataset.cpp

```diff
@@ -80,6 +80,11 @@ void bind_dataset_proxy_methods(py::class_<T, Ignored...> &c) {
   c.def("__contains__", &T::contains);
   c.def("copy", [](const T &self) { return Dataset(self); },
         "Return a (deep) copy.");
+  c.def_property_readonly(
+      "dims",
+      py::cpp_function([](T &self) { return self.dimensions(); },
+                       py::return_value_policy::move),
+      R"(Dict of dimensions.)");
```

Comment:
For `Variable` and `DataArray` `dims` just returns the labels. This feels inconsistent, should we consider a different method name?

Reply:
I guess there are 2 things.
1. in the python bindings, we could just make `.dims` return the labels
2. does `Dataset` really need an unordered map as `m_dims` in the core, or can it be a `Dimensions` object as it is for `Variable`?

I started having a quick look into 2, and thought it would take me a bit more than just half an hour, so I thought I would just have this PR as it is, and defer the other change to later.

Reply:
Re 2.: No, it should not. This was how it was in the past. I also causes troubles because it would limit the number of dimensions in a dataset.

Reply:
Yes regarding 1.) by the way, if dim labels is all you need.

Reply:
I made `.dims` return only the labels, and added `.shape` to get the sizes in each dim.

Reply:
Not sure about my implementation with `push_back` in a vector, maybe you have a more modern C++ way of writing this